p22/src/figures.rs

```diff
@@ -0,0 +1,160 @@
+#[derive(Debug, Clone)]
+pub struct Point {
+    x: f32,
+    y: f32,
+}
+
+#[derive(Debug, Clone)]
+pub struct Circle {
+    center: Point,
+    radius: f32,
+}
+
+#[derive(Debug, Clone)]
+pub struct Triangle {
+    points: [Point; 3],
+}
+
+#[derive(Debug, Clone)]
+pub struct Rectangle {
+    points: [Point; 4],
+}
+
+#[derive(Debug, Clone)]
+pub enum Shape {
+    Point,
+    Circle,
+    Triangle,
+    Rectangle,
+}
+
+pub trait Distance {
```

Comment:
You don't need to introduce traits in this exercise. You should just use free functions, i.e.:
```rust
fn circle_area(circle: &Circle) -> f32 {
    // ...
}

fn rectangle_area(circle: &Circle) -> f32 {
    // ...
}

/// etc
```